x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -1093,4 +1097,9 @@ export class AlertsClient {
 
     return fieldDescriptorToBrowserFieldMapper(fields);
   }
+
+  public async getAADFields({ ruleTypeId }: { ruleTypeId: string }) {
```

Comment:
Why is this async?

Reply:
Server error handling appears to expect async functions from the Alerts Client, it breaks when I don't make it async.

Reply:
You mean [this test](https://github.com/elastic/kibana/pull/158516/files#diff-da455d17e4cf4225546b9b031736c15d97e3cb7697d2ecd68e58b818ff844319R22-R43) is broken when not async? 
Just need to change `mockResolvedValue` by `mockReturnValue` and `mockRejectedValue` by `mockImplementation(() => { throw new Error() })`.
But anyway I guess that's not very important.